packages/trpc/server/routers/viewer/bookings.tsx

```diff
@@ -772,12 +772,8 @@ export const bookingsRouter = router({
     const isConfirmed = booking.status === BookingStatus.ACCEPTED;
     if (isConfirmed) throw new TRPCError({ code: "BAD_REQUEST", message: "Booking already confirmed" });
 
-    /** When a booking that requires payment its being confirmed but doesn't have any payment,
-     * we shouldn’t save it on DestinationCalendars
-     *
-     * FIXME: This can cause unintended confirmations on rejection.
-     */
-    if (booking.payment.length > 0 && !booking.paid) {
+    // If booking requires payment and is not paid, we don't allow confirmation
+    if (confirmed && booking.payment.length > 0 && !booking.paid) {
```

Comment:
It looks like previous code was not using confirmed input and now we using it to handle actions properly.